src/hiccup/compiler.clj

```diff
@@ -125,18 +171,21 @@
   [element]
   (let [[tag attrs content] (normalize-element element)]
     (if (container-tag? tag content)
-      (str "<" tag (render-attr-map attrs) ">"
-           (render-html content)
-           "</" tag ">")
-      (str "<" tag (render-attr-map attrs) (end-tag)))))
+      (build-string
+       "<" tag (render-attr-map attrs) ">"
+       (render-html content)
+       "</" tag ">")
```

Comment:
Maybe:

```clojure
      (build-string "<" tag (render-attr-map attrs) ">"
                    (render-html content)
                    "</" tag ">")
```

It makes the distinction between the two build-strings a little clearer. But this is just a stylistic suggestion, and won't affect whether I merge.

Reply:
431e741c7c213b3f459c2bb48935566e37ce7ba8